src/vec.cpp

```diff
@@ -1094,46 +1019,20 @@ void grid_volume::find_best_split(int desired_chunks, int &best_split_point,
         break;
     }
     int split_point = (first + last) / 2;
-    std::complex<double> costs = get_split_costs(d, split_point);
+    std::complex<double> costs = get_split_costs(d, split_point, fragment_cost);
     double left_cost = real(costs), right_cost = imag(costs);
     double total_cost = left_cost + right_cost;
-    double split_measure = max(left_cost * (desired_chunks - 1), right_cost);
+    double split_measure = max(left_cost / (desired_chunks / 2), right_cost / (desired_chunks - (desired_chunks / 2)));
+    if (d == longest_axis) split_measure *= 0.7;
     if (split_measure < best_split_measure) {
-      if (d == longest_axis || split_measure < (best_split_measure - (0.3 * best_split_measure))) {
-        // Only use this split_measure if we're on the longest_axis, or if the split_measure is
-        // more than 30% better than the best_split_measure. This is a heuristic to prefer lower
-        // communication costs when the split_measure is somewhat close.
-        // TODO: Use machine learning to get a cost function for the communication instead of hard
-        // coding 0.3
```

Comment:
Probably we should keep this comment (and put it before the `d == longest_axis` check).

Reply:
Fixed.